src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -249,13 +252,21 @@ public ClientBuilder codecDecorator(
     private Client newClient(Class<?> interfaceClass) {
         final Scheme scheme = Scheme.parse(uri.getScheme());
         final SessionProtocol sessionProtocol = scheme.sessionProtocol();
+        URI codecURI = this.uri;
 
-        final RemoteInvoker remoteInvoker = remoteInvokerFactory.getInvoker(sessionProtocol);
+        RemoteInvoker remoteInvoker = remoteInvokerFactory.getInvoker(sessionProtocol);
         if (remoteInvoker == null) {
             throw new IllegalArgumentException("unsupported scheme: " + scheme);
         }
 
-        final ClientCodec codec = createCodec(uri, scheme, interfaceClass);
+        String groupName = EndpointGroupUtil.getEndpointGroupName(this.uri.toString());
+        if (groupName != null) {
+            remoteInvoker = new EndpointGroupInvoker(remoteInvoker, groupName);
+            //TODO ugly,for RFC2616
+            codecURI = URI.create(codecURI.toString().replaceFirst(EndpointGroup.ENDPOINT_GROUP_MARK, ""));
```

Comment:
`codecURI.toString()` seems to be called more than once. Could you extract it into a local variable?